engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -333,6 +313,29 @@ private void addSkyNodes(RenderGraph renderGraph) {
         renderGraph.connect(lastUpdatedGBufferClearingNode, backdropNode, hazeIntermediateNode, hazeFinalNode);
     }
 
+    private void addWorldRenderingNodes(RenderGraph renderGraph) {
+        // Ideally, world rendering nodes only depend on gBufferClearingNode. However, since haze is produced
+        // by blurring the gBuffer and we don't want world objects/lighting to be a part of the haze,
+        // the world rendering nodes need to run after hazeIntermediateNode.
+        Node hazeIntermediateNode = renderGraph.findNode("engine:hazeIntermediateNode");
```

Comment:
Notice that while this is functionally correct, as mentioned in Slack I'd like the world-rendering nodes to be dependent on HazeFinalNode instead. 

I'd also reword the comment to:

```
/* Ideally, world rendering nodes only depend on the gBufferClearingNode. However, 
since the haze is produced by blurring the content of the gBuffer and we only want 
the sky color to contribute  to the haze, the world rendering nodes need to run 
after hazeFinalNode, so that the landscape and other meshes are not part of the haze.

Strictly speaking however, it is only the hazeIntermediateNode that should be processed
before the world rendering nodes. Here we have chosen to ensure also hazeFinalNode is
processed before the world rendering nodes not because it's necessary but to keep all
the haze-related nodes together.*/
```